### PR TITLE
AMDC REV F Hardware

### PR DESCRIPTION
**Pull request by DivyaMendpara**
This PR details modification made in [AMDC Rev F](https://github.com/Severson-Group/AMDC-Hardware/milestone/8) board

### Following changes were made for addressing issue #189 

- Part number of SCREW1...13 was changed to `36-9900-ND` 

image

### Following changes were made for addressing issue #227 

- As [suggested](https://github.com/Severson-Group/AMDC-Hardware/issues/227#issuecomment-1720226994) 1kOhm resistor is to be added to pull down PWM signals during boot-up. 
- As 1kOhm resistor was not present in previous revisions, @npetersen2 suggested to use two 470 ohm resistors in series. 
- R48 and R49 were added to the schematics as shown below and was integrated in PCB

image

### Following changes were made for addressing issue #226  

- As suggested by @npetersen2, [120 ohms four array resistor](https://www.digikey.com/en/products/detail/panasonic-electronic-components/EXB-38V121JV/256520)  was added and integrated into PCB

image

### Following changes were made for addressing issue #211  

- Footprint for AM26C32 (TSOP65P640X120-16N) was updated
- I have implemented the revised footprint across components U4, U6, U22, U33, U34, U46, U47, and U48 (as same footprints applies to all of them)

### CODE REVIEW

**Comment by npetersen2** [comment]
@codecubepi can you also review this PR?

- Open in Altium
- Verify changes to modified files
- Review each issue in the REV F milestone and add an issue comment per your review (see the REV E milestone and issue reviews as examples)
- Run checks to ensure design passes DRC for PCB and schematic review

**Review by npetersen2 — COMMENTED** [review]
@DivyaMendpara this is looking good!

I reviewed each issue in the milestone and added comments on the specific issues.

Overall, I updated the routing a little bit for the GPIO termination, and made sure the DRC passes and tried to help some of the soldermask sliver warnings specifically for the PicoZed footprint.

I think this is getting ready to order soon, pending @codecubepi and @elsevers review.

**Comment by npetersen2** [comment]
@DivyaMendpara please update the Project Parameters to change the Revision to be `F`.

You'll notice on the schematic sheets lower-right box it still says `E`. Once you update the Project Parameters, it should update on all sheets.

**Comment by codecubepi** [comment]
> @codecubepi can you also review this PR?
> 
> - Open in Altium
> - Verify changes to modified files
> - Review each issue in the REV F milestone and add an issue comment per your review (see the REV E milestone and issue reviews as examples)
> - Run checks to ensure design passes DRC for PCB and schematic review
> 

@npetersen2 

I will check it out!

**Comment by codecubepi** [comment]
@DivyaMendpara 

This looks pretty good to me! Thank you for addressing #211 for me!

I do have some questions on the routing for #226 :

Is there a reason this trace for GPIO4_IN3_P wraps around the via? Is this a length-match? If it's not, I think we could take the trace directly to the via (like the dashed red line). If it is a length-match, disregard.
image

Why is there a trace loop on GPIO1_IN2_N?
image


@npetersen2 

I went through all the issues under the [REV F milestone](https://github.com/Severson-Group/AMDC-Hardware/milestone/8) and added my comments. 

Questions:
- Don't differential signals need to be length/impedance matched usually? I'd like to confirm that the new traces to the termination resistors will not have any negative signal integrity effects.

- Do we need to update the [Revision Changes SchDoc](https://github.com/Severson-Group/AMDC-Hardware/blob/develop/altium/RevisionChanges.SchDoc) with info on REV F?
image

- I ran a DRC (cut-down version here: [AMDC_REV_F_DRC.pdf](https://github.com/Severson-Group/AMDC-Hardware/files/12809183/AMDC_REV_F_DRC.pdf)) and had a number of Solder Mask Sliver and Silk to Solder violations... anything to be concerned about here?


image

**Review by codecubepi — CHANGES_REQUESTED** [review]
Please see my above comments

**Comment by npetersen2** [comment]
@codecubepi thanks for the review! Regarding your comments:

1. @DivyaMendpara will address the routing issues for GPIO and the loop traces
2. Since GPIO is rather slow (10-20 or less MHz) and the physical length difference of the traces is reasonable small, I don't think length matching is required here. So, as long as the trace routing is "reasonable" i.e., no loops, no odd things, it should be fine.
3. Yes, the DRC has some rule violations for the solder mask and silkscreen. These are tolerable... ideally we would invest the time to update all "bad" footprints to reduce the DRC rule violations to zero, but I don't think this is worth the effort. In REV E, these violations existed and the assembly went without issue. The footprint updates (your issue and my changes to the PicoZed footprint) in REV F reduced the number of violations by 2x, so on the right path :) 

@DivyaMendpara once you have addressed the trace routing issues @codecubepi identified, please ask @codecubepi to review once again the routing. Then, upon approval, you can generate the `REVxxxxF/` design output folder and add it to git in this PR. We will then review the compiled output to make sure it is correct and then approve for ordering. Please follow the directions [here](https://github.com/Severson-Group/KnowledgeBase/blob/master/PCB-Design-Guidelines.md#exporting-design) for how to export the design.

**Review by codecubepi — DISMISSED** [review]
@DivyaMendpara 

Hey Divya, thank you for making the requested changes!

The fixes to the traces I mentioned look good. I did push one commit with a couple additional small fixes to squiggles in traces that I didn't mention in my previous review. I don't think these would have caused any problems other than bothering my OCD, so I just got them while I was reviewing your changes.

The additions to the RevisionChanges schematic page look good too!

I am ready to approve this.

**Comment by DivyaMendpara** [comment]
@npetersen2  and @codecubepi 
I created AMDC Rev F folder and uploaded design outputs (schematics, BOM, etc.) from Altium Designer. Please review and give your feedback.

File: REV20231005F/BOM_RevF.csv [hunk 1]
```diff
@@ -0,0 +1,71 @@
+Quantity,Supplier Part Number 1,Designator,Comment,Value,Supplier Order Qty 1
```
Comment (npetersen2) [inline REV20231005F/BOM_RevF.csv]: Can you update these columns to match those in the REV E BOM?

```
Quantity,Supplier Part Number 1,Designator,Comment,Value,Footprint
```

File: REV20231005F/Schematics_RevF.pdf
Comment (npetersen2) [inline REV20231005F/Schematics_RevF.pdf:1]: This is very minor, but can you add a space below the first line in the text box here:

image

Make it match the other text boxes on this page

File: REV20231005F/gerbers/PCB_v6-macro.APR_LIB
Comment (npetersen2) [inline REV20231005F/gerbers/PCB_v6-macro.APR_LIB:1]: Can you remove this file from the `gerbers/` folder

File: REV20231005F/gerbers/PCB_v6.REP
Comment (npetersen2) [inline REV20231005F/gerbers/PCB_v6.REP:1]: Remove this file

File: REV20231005F/gerbers/Status Report.Txt
Comment (npetersen2) [inline REV20231005F/gerbers/Status Report.Txt:1]: Remove this file

**Review by npetersen2 — CHANGES_REQUESTED** [review]
Thanks @DivyaMendpara , nearly there! I have a few comments.

- [x] Also, please generate the NC drill file (should have extension `.TXT`)

**Review by npetersen2 — CHANGES_REQUESTED** [review]
@DivyaMendpara @codecubepi 

Please add the README for the REV F folder

**Review by npetersen2 — CHANGES_REQUESTED** [review]
Please add the PicoZed to the BOM:

- Add a schematic symbol which "looks like" the PicoZed... maybe just a rectangle with the word "PicoZed" in it
- Mark as a `Mechanical` component (see the back page for other examples like the screws, stand-offs, heatsinks, jumpers, etc)
- Add the symbol to the back page with the correct part number. NOTE: the supplier for this **is not** Digi-Key
- Regenerate the BOM from the schematic view (following the instructions in the KB) and confirm it added a line for the PicoZed

https://docs.amdc.dev/hardware/subsystems/picozed.html

https://www.avnet.com/shop/us/products/avnet-engineering-services/aes-z7pz-7z030-som-i-g-rev-e-3074457345635303359

**Review by npetersen2 — COMMENTED** [review]
@DivyaMendpara which capacitor part number did you change in the previous commit?

**Comment by DivyaMendpara** [comment]
> @DivyaMendpara which capacitor part number did you change in the previous commit?

@npetersen2, I changed below capacitor value (now it is 0.1uF). Total qty (40+45).
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Patron/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Patron/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">



</head>

<body link="#0563C1" vlink="#954F72">



45 | 1276-1012-6-ND | C6, C7, C9, C10, C11, C12, C14, C15, C16,   C17, C19, C20, C21, C23, C31, C32, C33, C34, C65, C66, C67, C68, C69, C70,   C92, C93A, C93B, C93C, C93D, C93E, C93F, C93G, C93H, C94A, C94B, C94C, C94D,   C94E, C94F, C94G, C94H, C118A, C118B, C119A, C119B | Capacitor | 0.1uF | 0603-CAP
-- | -- | -- | -- | -- | --
40 | 1276-1012-6-ND | C24, C25, C27, C28, C29, C30, C71A, C71B, C71C, C71D, C71E, C71F, C73A,   C73B, C73C, C73D, C73E, C73F, C74A, C74B, C74C, C74D, C74E, C74F, C76A, C76B,   C76C, C76D, C76E, C76F, C99, C100, C101, C102, C107, C110, C111, C112, C113,   C114 | Capacitor | 100nF | 0603-CAP



</body>

</html>

**Review by npetersen2 — COMMENTED** [review]
Looks good to me! I approve. Thanks @DivyaMendpara for your hard work on pushing this through :)

@elsevers can you review the `Files changed` and approve? We will get this merged and proceed with ordering boards. Thanks

File: REV20231005F/BOM_RevF.csv [hunk 2]
```diff
@@ -0,0 +1,73 @@
+Quantity,Supplier Part Number 1,Designator,Comment,Value,Footprint
+
+"48","507-0ZCM0004FF2GCT-ND","RT1A, RT1B, RT1C, RT1D, RT2A, RT2B, RT2C, RT2D, RT3A, RT3B, RT3C, RT3D, RT4A, RT4B, RT4C, RT4D, RT5A, RT5B, RT5C, RT5D, RT6A, RT6B, RT6C, RT6D, RT7A, RT7B, RT7C, RT7D, RT8A, RT8B, RT8C, RT8D, RT9A, RT9B, RT9C, RT9D, RT10A, RT10B, RT10C, RT10D, RT11A, RT11B, RT11C, RT11D, RT12A, RT12B, RT12C, RT12D","0ZCM0004FF2G","","PTCC1607X06N"
+"1","68AH8094","PICOZED1","7Z030-PicoZed","",""
```
Comment (elsevers) [inline REV20231005F/BOM_RevF.csv]: It looks like we don't have the Picozed's part number recorded correctly. Where is `68AH8094` coming from? I don't get anything if I search Octopart for it.

The [AMDC Docs](https://docs.amdc.dev/hardware/subsystems/picozed.html) provides a link to the [`7Z030 version of the Picozed`](https://www.avnet.com/shop/us/products/avnet-engineering-services/aes-z7pz-7z030-som-i-g-rev-e-3074457345635303359), which indicates a part number of `AES-Z7PZ-7Z030-SOM-I-G/REV-E`. Seems like that is what we should be using?

**Review by elsevers — CHANGES_REQUESTED** [review]
Thanks @DivyaMendpara, @npetersen2, and @codecubepi.

It looks like there is one small hiccup with the Picozed partnumber -- see my inline comment.

File: REV20231005F/BOM_RevF.csv (hunk 2 above)
Comment (DivyaMendpara) [reply REV20231005F/BOM_RevF.csv]: Hello @elsevers, I updated Picozed part number as per your suggestion. Please review and give your feedback

**Review by elsevers — APPROVED** [review]
I see the updated Picozed. This looks good to me. Thanks @DivyaMendpara

**Review by npetersen2 — APPROVED** [review]
Thanks @DivyaMendpara !

**Review by codecubepi — APPROVED** [review]
Looks good, thanks @DivyaMendpara !

**Comment by codecubepi** [comment]
@npetersen2 @DivyaMendpara 

It looks like the following issues in this repository were never closed after this was merged:

- #189 
- #211
- #226 
- #227 

Can we go ahead and close these now?

**Comment by npetersen2** [comment]
Thanks @codecubepi , I went ahead and closed the issues you linked above.

**Pull request merged**